base/float.jl

```diff
@@ -129,13 +131,110 @@ function convert(::Type{Float32}, x::Int128)
     reinterpret(Float32, s | d + y)
 end
 
+function convert(::Type{Float16}, val::Float32)
+    f = reinterpret(UInt32, val)
+    i = (f >> 23) & 0x1ff + 1
+    sh = shifttable[i]
+    f &= 0x007fffff
+    h::UInt16 = basetable[i] + (f >> sh)
+    # round
+    # NOTE: we maybe should ignore NaNs here, but the payload is
+    # getting truncated anyway so "rounding" it might not matter
+    nextbit = (f >> (sh-1)) & 1
+    if nextbit != 0
+        if h&1 == 1 ||  # round halfway to even
+            (f & ((1<<(sh-1))-1)) != 0  # check lower bits
+            h += 1
```

Comment:
The comments are helpful but IMHO splitting the condition like that makes it harder to read. Could maybe do something like
```julia
# Round halfway to even, or check lower bits
if h & 1 == 1 || (f & ((1 << (sh - 1)) - 1)) != 0
    h += 1
end
```
?